cancer_ai/validator/utils.py

```diff
@@ -84,3 +92,141 @@ def get_competition_config(path: str) -> CompetitionsListModel:
+    bt.logging.trace(f"Fetching organization data references from Hugging Face repo {hf_repo_id}")
+
+    # prevent stale connections
+    custom_headers = {"Connection": "close"}
+
+    files = hf_api.list_repo_tree(
+        repo_id=hf_repo_id,
+        repo_type="space",
+        token=hf_token,
+        recursive=True,
+        expand=True,
+    )
+
+    yaml_data = []
+
+    for file_info in files:
+        if file_info.__class__.__name__ == 'RepoFile':
+            file_path = file_info.path
+
+            if file_path.startswith('datasets/') and file_path.endswith('.yaml'):
+                local_file_path = hf_hub_download(
+                    repo_id=hf_repo_id,
+                    repo_type="space",
+                    token=hf_token,
+                    filename=file_path,
+                    headers=custom_headers,
+                )
+
+                last_commit_info = file_info.last_commit
+                commit_date = last_commit_info.date if last_commit_info else None
+
+                if commit_date is not None:
+                    date_uploaded = commit_date
+                else:
+                    bt.logging.warning(f"Could not get the last commit date for {file_path}")
+                    date_uploaded = None
+
+                with open(local_file_path, 'r') as f:
+                    try:
+                        data = yaml.safe_load(f)
+                    except yaml.YAMLError as e:
+                        bt.logging.error(f"Error parsing YAML file {file_path}: {str(e)}")
+                        continue  # Skip this file due to parsing error
+
+                yaml_data.append({
+                    'file_name': file_path,
+                    'yaml_data': data,
+                    'date_uploaded': date_uploaded
+                })
+        else:
+            continue
+    return yaml_data
+
+async def fetch_yaml_data_from_local_repo(local_repo_path: str) -> list[dict]:
+    """
+    Fetches YAML data from all YAML files in the specified local directory.
+    Returns a list of dictionaries containing file name, YAML data, and the last modified date.
+    """
+    yaml_data = []
+
+    # Traverse through the local directory to find YAML files
+    for root, _, files in os.walk(local_repo_path):
+        for file_name in files:
+            if file_name.endswith('.yaml'):
+                file_path = os.path.join(root, file_name)
+                relative_path = os.path.relpath(file_path, local_repo_path)
+                commit_date = datetime.fromtimestamp(os.path.getmtime(file_path))
+
+                with open(file_path, 'r') as f:
+                    data = yaml.safe_load(f)
+
+                yaml_data.append({
+                    'file_name': relative_path,
+                    'yaml_data': data,
+                    'date_uploaded': commit_date
+                })
+
+    return yaml_data
+
+async def get_new_organization_data_updates(fetched_yaml_files: list[dict]) -> list[DatasetReference]:
```

Comment:
you shouldn't name variable with `yaml`, maybe `fetched_reference_config`, it is a dict here not yaml file